### PR TITLE
Issue 13494

### PR DESCRIPTION
**Pull request by crshnburn**
Update the DefaultGenerator to not remove server URLs that are a single `/`

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by crshnburn** [comment]
@wing328 I've rebased and fixed the merge issues.

Spec document showing the issue can be found [here](https://raw.githubusercontent.com/zosconnect/sample-cics-api/main/start/src/main/api/openapi.yaml)

**Comment by crshnburn** [comment]
@wing328 Is it possible to get this PR merged?

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

File: samples/client/petstore/groovy/bin/main/org/openapitools/api/PetApi.groovy [hunk 1]
```diff
@@ -214,7 +214,7 @@ class PetApi {
         contentType = 'multipart/form-data';
         bodyParams = [:]
         bodyParams.put("additionalMetadata", additionalMetadata)
-        bodyParams.put("file", file)
+        bodyParams.put("file", _file)
```
Comment (wing328) [inline samples/client/petstore/groovy/bin/main/org/openapitools/api/PetApi.groovy:217]: i don't understand why this change occurs.

maybe merging the latest master into your branch and regenerate the samples to see if this change goes away?

File: samples/server/petstore/rust-server/output/multipart-v3/src/client/mod.rs [hunk 2]
```diff
@@ -396,7 +396,7 @@ impl<S, C> Api<C> for Client<S, C> where
     {
         let mut client_service = self.client_service.clone();
         let mut uri = format!(
-            "{}/multipart_related_request",
+            "{}//multipart_related_request",
```
Comment (wing328) [inline samples/server/petstore/rust-server/output/multipart-v3/src/client/mod.rs:399]: i think this change may break rust server but I'm no expert in rust

**Comment by wing328** [comment]
according to the spec (partially shown below):
```
openapi: 3.0.0
info:
  title: CatalogManagerApi
  description: Manage a catalog through an API for CICS.
  version: '1.1'
  license:
    name: Apache-2.0
    url: https://opensource.org/licenses/Apache-2.0
servers:
- url: /
security:
  - BasicAuth: []
  - BearerAuth: []
```

should the `servers` entry contains the host name so that the client knows which server to make the request?

**Comment by wing328** [comment]
closing it as there's no update.
